core/src/main/java/hudson/cli/InstallPluginCommand.java

```diff
@@ -80,6 +82,19 @@ protected int run() throws Exception {
         }
 
         for (String source : sources) {
+            if (source.equals("=")) {
+                if (name == null) {
+                    throw new IllegalArgumentException("-name required when using -source -");
```

Comment:
I interpreted the usage doc as if name is not specified then Short-Name would be read: _"Otherwise the name is assumed to be the short name of the plugin..."_

Reply:
#4123